### DISCUSSION TITLE
`cursor_by` with descending order

### DISCUSSION THREAD

**Discussion by uedaeita** [SeaQL/sea-orm#1711] [Q&A] [answered]
How should I generate a cursor for a column with descending order by default?

For example

```rust
let cakes  = cake::Entity::find()
    .cursor_by(cake::Column::Id)
    .order_by_desc(cake::Column::Id)
    .all(db)
    .await?;
```

This creates a query like `SELECT * FROM cakes ORDER BY id ASC;` but I want `SELECT * FROM cakes ORDER BY id DESC;`.

**Comment by billy1624** [comment, marked as answer]
Hey @uedaeita, you can use the `last` method to paginate the result in descending order.
https://docs.rs/sea-orm/latest/sea_orm/struct.Cursor.html#method.last

**Reply by uedaeita** [reply]
@billy1624 
Thanks for your reply!

As your suggestion I was able to generate a query as `DESC` but the order of the actual result is in ascending order. I ended up using `reverse()` with the result.

I suppose this is the current spec but less intuitive? 🤔 
https://www.sea-ql.org/SeaORM/docs/next/basic-crud/select/#cursor-pagination

**Reply by billy1624** [reply]
Right, that's the expected behaviour
